test/gpu_test.cpp

```diff
@@ -160,3 +163,105 @@ TEST(gpu_test, memory_manager_seperate_workspaces_allocsize)
     emitter.allocate_primitive_memory();
     EXPECT_EQ(emitter.sizeof_device_allocation(), total_size);
 }
+
+TEST(gpu_test, topk_3d_global_mem)
+{
+    Shape shape{4, 8192, 5};
+    auto A_interp = make_shared<op::Parameter>(element::f32, shape);
+    auto A_gpu = make_shared<op::Parameter>(element::f32, shape);
+
+    auto B_interp = make_shared<op::TopK>(A_interp, 1, element::i32, 10, true);
+    auto B_gpu = make_shared<op::TopK>(A_gpu, 1, element::i32, 10, true);
+
+    auto interp_f_0 = make_shared<Function>(make_shared<op::GetOutputElement>(B_interp, 0),
+                                            op::ParameterVector{A_interp});
+    auto interp_f_1 = make_shared<Function>(make_shared<op::GetOutputElement>(B_interp, 1),
+                                            op::ParameterVector{A_interp});
+
+    auto gpu_f_0 = make_shared<Function>(make_shared<op::GetOutputElement>(B_gpu, 0),
+                                         op::ParameterVector{A_gpu});
+    auto gpu_f_1 = make_shared<Function>(make_shared<op::GetOutputElement>(B_gpu, 1),
+                                         op::ParameterVector{A_gpu});
+
+    vector<vector<float>> args_index;
+    for (shared_ptr<op::Parameter> param : interp_f_0->get_parameters())
+    {
+        vector<float> tensor_val(shape_size(param->get_shape()));
+        iota(tensor_val.begin(), tensor_val.end(), 0.0f);
+        args_index.push_back(tensor_val);
+    }
+
+    auto interp_results_0 = execute<float, int32_t>(interp_f_0, args_index, "INTERPRETER");
+    auto gpu_results_0 = execute<float, int32_t>(gpu_f_0, args_index, "GPU");
```

Comment:
Sorry, I can't see any GPU specific things in these tests. Why don't add them to a common place for all backends?

Reply:
topk_3d_global_mem test is a case specific to nvGPUs wherein memory is allocated on the gpu global memory as compared to the shared memory on the GPU when the size of a row exceeds a particular limit. topk_fanout_graph_transform test locks the behavior wherein reshape ops are added before and after topk. This is also specific to nvGPU backend. These tests would definitely pass for all backends but I added these to gpu_test.cpp since these tests are intended to test changes specific to nvGPU backends. I can move them to backend_topk.in.cpp if you think that is appropriate.

Reply:
moved to general tests